crates/filesystem/src/lib.rs

```diff
@@ -0,0 +1,11 @@
+pub trait FileSystem {
+    fn add_file(&mut self, path: &str, contents: &str) -> Result<(), String>;
+    fn remove_file(&mut self, path: &str) -> Result<(), String>;
+    fn read_file(&self, path: &str) -> Result<String, String>;
```

Comment:
What about making these async fn? We may want to have `S3` filesystem or something like network read and we will want them to be async.

Reply:
Done